### PR TITLE
Tooltip: make widget accessible

### LINKED ISSUE #42383: Refactor Tooltip to be accessible
Refactor `Tooltip` to be accessible

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR makes the tooltip widget to be accessible. According to the [W3G](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html), tooltips are expected to satisfy certain criteria. This PR achieves that.
**Which issue(s) this PR fixes**:

Fixes #42383

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -16,6 +16,19 @@ export type PopoverContent = string | React.ReactElement<any> | ((props: Popover
 export const Tooltip: FC<TooltipProps> = React.memo(({ children, theme, ...controllerProps }: TooltipProps) => {
   const tooltipTriggerRef = createRef<HTMLElement | VirtualElement>();
   const popperBackgroundClassName = 'popper__background' + (theme ? ' popper__background--' + theme : '');
+  const closePopover = (event: React.KeyboardEvent<HTMLDivElement>, hidePopper: () => void) => {
```
Comment (thisisobate) [inline packages/grafana-ui/src/components/Tooltip/Tooltip.tsx:19]: Noticed this piece of code is duplicated across the codebase. Would extract this into an internal util function and reuse. This would be done in a separate PR though

**Review by joshhunt — APPROVED** [review]

**Comment by thisisobate** [comment]
woow! @joshhunt that was instant! 😅🚀🔥
